File: Katiba55.API/Entities/Work.cs
namespace Katiba55.API.Entities
{
    public class Work: BaseEntity
    {
        public string Name { get; set; }
        public DateTime? EstimatedStartDate { get; set; }
        public DateTime? EstimatedEndDate { get; set; }
        public DateTime? ActualStartDate { get; set; }
        public DateTime? ActualEndDate { get; set; }
        public double? TotalValue { get; set; }
        public double? ExecutedValue { get; set; }
        public double? RemainingValue { get; set; }
        public double? RelativeWeightPercent { get; set; } 
        public double? ExecutionPercent { get; set; }
        public DateTime? ExecutionDate { get; set; }
        public ExecutionStatus ExecutionStatus { get; set; } 
        public string? Notes { get; set; }

        public int ProjectId { get; set; }
        public Project Project { get; set; }

        public ICollection<Item> Items { get; set; }
        public ICollection<BOQ> BOQs { get; set; }
        public ICollection<WorkExecutionHistory>? ExecutionHistories { get; set; }
        public ICollection<WorkCompany> WorkCompanies { get; set; }
    }
}

File: Katiba55.Client/src/app/models/works/work.ts
import { ExecutionStatus } from "../../enums/execution-status.enum";

export interface Work {
    id: number,
    name: string,
    startDate?: Date,
    endDate?: Date,
    estimatedStartDate?: Date
    estimatedEndDate?: Date
    actualStartDate?: Date
    actualEndDate?: Date
    totalValue?: number,
    executedValue?: number,
    relativeWeightPercent?: number,
    executionPercent?: number,
    relativeExecutionPercent?: number,
    executionDate?: Date,
    executionStatus: ExecutionStatus,
    responsibleId?: number,
    projectId: number,
    notes?: string,
}
